src/screens/HomeScreens/components/InfoTable/InfoTable.js

```diff
@@ -1,78 +1,157 @@
         delInfo(id)
+        setCheckNum(pre=>pre-1)
+        console.log(checkNum)
+        toast('Successfully deleted one todo task')
     }
     const checkedItemHandler=(event)=>{
-        var isChecked=event.target.checked;   
+        var isChecked=event.target.checked;  
+        isChecked ? setCheckNum(pre=>pre+1) : setCheckNum(pre=>pre-1) 
         var infoid=parseInt(event.target.value);
         changeCheckedStatus({isChecked:isChecked,id:infoid})
         // dispatch(changeCheckedstatusAction({isChecked:isChecked,id:infoid}))
         setSelectedChecked(!selectedChecked)
+
     }
     const selectAllHandler=(event)=>{
         // dispatch(changeAllCheckedstatusAction(event.target.checked))
         // allinfo.map(item=>item.isChecked=event.target.checked)
+        var isChecked=event.target.checked
+        isChecked ? setCheckNum(infos.length) : setCheckNum(0)
         selectAll(event.target.checked)
         setAllChecked(!allChecked)
+        
     }
     
     const deleteSelectedInfoHandler=()=>{
         // dispatch(delSelectedInfoAction())
         delSeleted()
         setAllChecked(false)
         setSelectedChecked(false)
+        setCheckNum(0)
+        console.log(checkNum)
+        toast('Successfully deleted selected todo task')
     }
+    const sortHandler=()=>{
+        if(sortType===0)
+        {
+            infos.sort((a,b)=>{return a.deadline-b.deadline})
+            setSortType(1)
+        }
+        else {
+            infos.sort((a,b)=>{return b.deadline-a.deadline})
+            setSortType(0)
+        }
+        
+        
+    }
+    const filterHandler=(e)=>{
+        if(e.target.value === ""){
+            setAllinfos(infos)
+        }
+        else {
+            setAllinfos(infos.filter(item=>item.category===e.target.value))
+        }
+        
+    }
+
+    useEffect(() => {
+        setAllinfos(infos)
+        if(checkNum ===0){
+            setAllChecked(false)
+        }
+        else {
+            if(checkNum===infos.length){
+                setAllChecked(true)
+            }
+            else {
+                setAllChecked(false)
+            }
+        }
+
+        
+    }, [checkNum, infos])
 
     return (
         <Container>
-        <h3>Todo List</h3>
-        <br />
-        <div>
-        <Button variant='secondary' onClick={deleteSelectedInfoHandler}>Delete seleted</Button>
-        </div>
-        <br />
-        <Table className="table table-hover">
+            <h3>Todo List</h3>
+            <br />
+            {(infos.length!=0) && 
+            <div>
+            <Row>
+                <Col md={4}>
+                    <Button variant='secondary' disabled={(checkNum===0)? true : false} onClick={deleteSelectedInfoHandler}>Delete seleted</Button> 
+                </Col>
+                <Col md={{ span: 4, offset: 4 }}>
+                    <label >Filter by category: &nbsp;</label>
+                    <select name="category" onChange={filterHandler}> 
```

Comment:
I could see duplicated codes for this select options. Consider consolidating this part with `TodoSelect.js`

Reply:
done